elytron/src/main/java/org/wildfly/extension/elytron/HttpAuthenticationFactoryService.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2019 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wildfly.extension.elytron;
+
+import static org.wildfly.common.Assert.checkNotNullParam;
+
+import java.util.concurrent.Callable;
+import java.util.function.Supplier;
+
+import org.jboss.msc.service.Service;
+import org.jboss.msc.service.StartContext;
+import org.jboss.msc.service.StartException;
+import org.jboss.msc.service.StopContext;
+import org.wildfly.security.auth.server.HttpAuthenticationFactory;
+
+/**
+ * A very trivial {@link Service} implementation where creation of the value type can easily be wrapped using a {@link Supplier}
+ * and the scheduled executor in HttpAuthenticationFactory is shut down.
+ *
+ * @author <a href="mailto:aabdelsa@gmail.com">Ashley Abdel-Sayed</a>
+ */
+
+class HttpAuthenticationFactoryService implements Service {
+
+    private Callable<HttpAuthenticationFactory> httpAuthenticationFactorySupplier;
```

Comment:
This should be 'final' with the supplier passed into the constructor and the setter method removed. The no-arg constructor should be removed as well.